contrib/generic_proxy/filters/network/source/proxy.cc

```diff
@@ -314,6 +316,12 @@ void UpstreamManagerImpl::onPoolFailureImpl(ConnectionPool::PoolFailureReason re
   parent_.onBoundUpstreamConnectionEvent(Network::ConnectionEvent::RemoteClose);
 }
 
+void UpstreamManagerImpl::onDecodingSuccess(ResponsePtr response, ExtendedOptions options,
+                                            bool end_stream) {
+  ENVOY_LOG(debug, "response end stream[{}]", end_stream);
+  onDecodingSuccess(std::move(response), options);
+}
```

Comment:
So, the shared upstream implementation couldn't be used for multiple frames response.

Reply:
We traced the call stack and the response was just processed by  `UpstreamRequest::onDecodingSuccess(ResponsePtr response, ...)` and currently, it is seem not to use `UpstreamManagerImpl::onDecodingSuccess(ResponsePtr response, ...)`. 

![image](https://github.com/envoyproxy/envoy/assets/39832249/930ec2c4-b760-422c-81d8-161ca58f452c)

In addition, what is the difference between them and how to use  `UpstreamManagerImpl::onDecodingSuccess(ResponsePtr response, ...)` @wbpcode

Reply:
I will move the `UpstreamManagerImpl` to the `router` in next PR. This is a special feature to bound upstream connection and downstream connection.

If a special flag is set, when first request is handled, the upstream connection will be bound to the downstream connection. And all following requests will be sent to this same upstream connection. This is useful for protocol with stateful connection. You can just ignore it if you have no similar scenario.